### PR TITLE
BucketId Dynamic Metadata for RateLimitQuota filter

### PR DESCRIPTION
**Pull request by bsurber**
Commit Message: Add dynamic metadata to the rate_limit_quota filter to export the selected bucket for a given request for logging
Additional Description: Functionality tested using an access log filter in integration testing
Risk Level: low
Testing: integration testing
Docs Changes:
Release Notes:
Platform Specific Features:

### CODE REVIEW

**Comment by bsurber** [comment]
The presubmit failure seems to have come from a failure to cache the results of the presubmit, not an actual failing of a test

File: source/extensions/filters/http/rate_limit_quota/filter.cc [hunk 1]
```diff
@@ -28,27 +28,35 @@ Http::FilterHeadersStatus RateLimitQuotaFilter::decodeHeaders(Http::RequestHeade
...
   ENVOY_LOG(trace, "Generated the associated hashed bucket id: {} for bucket id proto:\n {}",
             bucket_id, bucket_id_proto.DebugString());
+
+  ProtobufWkt::Struct bucket_log;
+  auto* bucket_log_fields = bucket_log.mutable_fields();
+  for (const auto& bucket : bucket_id_proto.bucket())
+    (*bucket_log_fields)[bucket.first] = ValueUtil::stringValue(bucket.second);
+
+  callbacks_->streamInfo().setDynamicMetadata(
+      "envoy.extensions.http_filters.rate_limit_quota.bucket", bucket_log);
```
Comment (tyxia) [inline source/extensions/filters/http/rate_limit_quota/filter.cc:50]: nit: Can you define a string const for key `"envoy.extensions.http_filters.rate_limit_quota.bucket"`?

But it is really up to you.

**Review by tyxia — DISMISSED** [review]
Great work! Thanks

Let's wait for the CI to be back so that I can merge this commit.

File: source/extensions/filters/http/rate_limit_quota/filter.cc (hunk 1 above)
Comment (bsurber) [reply source/extensions/filters/http/rate_limit_quota/filter.cc]: Updated.

**Review by tyxia — APPROVED** [review]
Thanks!

**Pull request merged**